src/Quaternion.jl

```diff
@@ -47,17 +47,47 @@ julia> quat([1, 2, 3])
  Quaternion{Int64}(3, 0, 0, 0)
 ```
 """
-quat
+function quat end
 
-quat(p, v1, v2, v3) = Quaternion(p, v1, v2, v3)
-quat(x) = Quaternion(x)
```

Comment:
Technically this removal would be breaking. e.g. this would block users from constructing quaternions with `SymbolicUtils.Num` or `Unitful.Quantity`. Not that I'm opposed to it, but we need to be certain we're fine with that first.

Reply:
I think this is not a breaking change.

No problem with `Symbolics.Num` because it's a subtype of `Real`.

```julia
julia> using Symbolics, Quaternions

julia> a = Symbolics.Num(3)
3

julia> quat(a,a,a,a)
Quaternion{Num}(3, 3, 3, 3)

julia> quat(a)
Quaternion{Num}(3, 0, 0, 0)

julia> Symbolics.Num <: Real
true
```

`Unitful.Quantitiy` is already not compatible with Quaternions.jl.

**Before this PR**
```julia
julia> using Unitful, Quaternions

julia> d = 3u"°"
3°

julia> quat(d)
ERROR: MethodError: no method matching Quaternion(::Quantity{Int64, NoDims, Unitful.FreeUnits{(°,), NoDims, nothing}})

Closest candidates are:
  (::Type{T})(::T) where T<:Number
   @ Core boot.jl:792
  (::Type{T})(::AbstractChar) where T<:Union{AbstractChar, Number}
   @ Base char.jl:50
  (::Type{T})(::Base.TwicePrecision) where T<:Number
   @ Base twiceprecision.jl:266
  ...
```

**After this PR**
```julia
julia> using Unitful, Quaternions

julia> d = 3u"°"
3°

julia> quat(d)
ERROR: MethodError: no method matching quat(::Quantity{Int64, NoDims, Unitful.FreeUnits{(°,), NoDims, nothing}})

Closest candidates are:
  quat(::Quaternion)
   @ Quaternions ~/.julia/dev/Quaternions/src/Quaternion.jl:52
  quat(::Real)
   @ Quaternions ~/.julia/dev/Quaternions/src/Quaternion.jl:53
  quat(::Real, ::Real, ::Real, ::Real)
   @ Quaternions ~/.julia/dev/Quaternions/src/Quaternion.jl:54
  ...
```